src/app/routes/registration/enrolment-form/enrolment-form.component.spec.ts

```diff
@@ -119,6 +119,62 @@ describe('EnrolmentFormComponent', () => {
       expect(onChain.nativeElement.checked).toBeTruthy('on chain should be selected');
       expect(submit.nativeElement.disabled).toBeFalsy('submit button should be enabled');
     });
+
+    it('should disable checkboxes when only onChain is predefined and set to true', () => {
+      component.fieldList = [];
+      component.predefinedRegTypes = {onChain: true};
+
+      fixture.detectChanges();
+
+      const {offChain, onChain} = getSelectors(hostDebug);
+
+      expect(offChain.nativeElement.checked).toBeFalsy('off chain should be deselected');
+      expect(offChain.nativeElement.disabled).toBeTruthy('off chain is not disabled');
+      expect(onChain.nativeElement.checked).toBeTruthy('on chain should be selected');
+      expect(onChain.nativeElement.disabled).toBeTruthy('on chain is not disabled');
+    });
+
+    it('should disable checkboxes when only offChain is predefined and set to true', () => {
+      component.fieldList = [];
+      component.predefinedRegTypes = {offChain: true};
+
+      fixture.detectChanges();
+
+      const {offChain, onChain} = getSelectors(hostDebug);
+
+      expect(offChain.nativeElement.checked).toBeTruthy('off chain should be selected');
+      expect(offChain.nativeElement.disabled).toBeTruthy('off chain is not disabled');
+      expect(onChain.nativeElement.checked).toBeFalsy('on chain should be deselected');
+      expect(onChain.nativeElement.disabled).toBeTruthy('on chain is not disabled');
+    });
+
+    it('should disable checkboxes when offChain and onChain are predefined and set to true', () => {
```

Comment:
Should this say `enable`?
```suggestion
    it('should enable checkboxes when offChain and onChain are predefined and set to true', () => {
```

Reply:
No, what i understood checkboxes should be disabled when are predefined by url. So if we have in url onchain=true then checkbox for on-chain should be disabled but checked. Checkbox for off-chain should be also disabled but unchecked (disabled means when i click on it, nothing happens)

Reply:
Oh yes, you are right 👍